eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -132,6 +133,31 @@ jobs:
   - checkout: self
     submodules: recursive
 
+  # Validate all the files are formatted correctly according to the
+  # .clang-format file. This step is macOS only to use clang-format-11
+  - bash: |
+      # Run clang-format recursively on each source and header file within the repo sdk folder.
+      echo "Check clang-formatting"
+      brew install clang-format
+      clang-format --version
+      find sdk/ -regex '.*\.\(cpp\|hpp\)' -exec clang-format -style=file -i {} \;
+
+      if [[ `git status | grep modified | awk '{print $2}'` ]]; then
+        echo Some files were not formatted correctly according to the .clang-format file.
+        echo Please run clang-format version 10 or greater to fix the issue by using this bash command at the root of the repo:
```

Comment:
Update the contributing guide doc to say we need clang-format 10 or higher.

https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#third-party-dependencies
![image](https://user-images.githubusercontent.com/6527137/105080780-2b8fff80-5a46-11eb-81e0-e95f2212668f.png)